lib/cdo/script_constants.rb

```diff
@@ -51,14 +56,43 @@ def ScriptConstants.hoc?(name)
         MINECRAFT_NAME,
         INFINITY_NAME,
         ARTIST_NAME
-    ].include? name
-  end
+    ],
+    csf: [
+        TWENTY_HOUR_NAME,
+        COURSE1_NAME,
+        COURSE2_NAME,
+        COURSE3_NAME,
+        COURSE4_NAME
+    ],
+    csp: [
+        CSP_UNIT1_NAME,
+        CSP_UNIT2_NAME,
+        CSP_UNIT3_NAME,
+        CSP_UNIT4_NAME,
+        CSP_UNIT5_NAME,
+        CSP_UNIT6_NAME
+    ],
+    math: [
+      ALGEBRA_NAME,
+      ALGEBRA_A_NAME,
+      ALGEBRA_B_NAME
+    ],
+    twenty_hour: [TWENTY_HOUR_NAME],
+    flappy: [FLAPPY_NAME],
+    minecraft: [MINECRAFT_NAME]
+  }
 
-  def ScriptConstants.flappy?(name)
-    name == FLAPPY_NAME
+  def self.method_missing(symbol, *args)
+    method = symbol.to_s
+    category = method[0..-2].to_sym
+    return CATEGORIES[category].include? args.first if args.length == 1 &&
+      method[-1] == '?' &&
+      CATEGORIES.include?(category)
+    super
   end
```

Comment:
I replaced all the existing category methods like `hoc?` and `twenty_hour?` with this `method_missing` definition. It just uses the Category definitions above. Added a comment.